src/Game.js

```diff
@@ -0,0 +1,78 @@
+function Game() {
+  this.frames = [[[]]];
+  this.currentFrame = 0;
+}
+
+Game.prototype.addScore = function(score) {
+  if (this.currentFrame === 9 && this.frames[this.currentFrame].length >= 1) {
+    this._tenthFrame(score);
+  } else {
+    this._checkFrame();
+    this._currentFrameScore().push(score);
+    this._applyBonus(score);
+    this._checkBonus(score);
+  }
+};
+
+Game.prototype.frameTotal = function() {
+ return this._currentFrameScore().reduce(function(a, b) {
+    return a + b;
+  }, 0);
+};
+
+Game.prototype.total = function() {
+  var total = 0;
+  for (var i = 0; i < this.frames.length; i++) {
```

Comment:
Having a separate frame object may clean up this code.  The Frame object can calculate the total for a frame and then you can reduce over that.